File: 937-online-stock-span/online-stock-span.java
class StockSpanner {

    ArrayList<Integer> no=new ArrayList<>();
    ArrayList<Integer> hi=new ArrayList<>();
    public StockSpanner() {
        
    }
    
    public int next(int price) {
        int s=no.size();
        no.add(price);
        if(s==0){
          hi.add(-1);
          return 1;
        }
        else if(no.get(s-1)>price){
            hi.add(s-1);
            return 1;
        }
        else{
            int c=1;
            int i=s-1;
            while(no.get(i)<=price){   
                c=c+1;
                i=hi.get(i);
                if(i==-1) break;
            }
            hi.add(i);
            return s-i;
        }
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */

File: 937-online-stock-span/online-stock-span.py
class StockSpanner(object):

    def __init__(self):
        self.stack = []
        

    def next(self, price):
        """
        :type price: int
        :rtype: int
        """
        count = 1
        while len(self.stack) != 0 and self.stack[-1][0] <= price:
            _, pre = self.stack[-1]
            count += pre
            self.stack.pop(-1)
        self.stack.append((price, count))
        return count
        


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)